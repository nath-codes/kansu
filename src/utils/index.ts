import { WindowLocation } from 'reach__router'

const getPageNav = (location: WindowLocation | undefined, urls: string[]) => {
  if (location && location.pathname) {
    const currentIndex = urls.findIndex(
      (url: string) => url === location.pathname
    )
    const lastIndex = urls.length - 1
    const nextPage =
      currentIndex === lastIndex ? undefined : urls[currentIndex + 1]
    const prevPage = currentIndex === 0 ? undefined : urls[currentIndex - 1]

    return { nextPage, prevPage }
  }
}

export default getPageNav
