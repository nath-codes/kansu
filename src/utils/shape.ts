import {
  ANIMATION_MAX_DIMENSION,
  COLOR_DARK_BLUE_RGBA,
  COLOR_GREY_RGBA,
  COLOR_WHITE_RGBA,
} from '../constants'

const colors: [number[], number[], number[]] = [
  COLOR_DARK_BLUE_RGBA,
  COLOR_GREY_RGBA,
  COLOR_WHITE_RGBA,
]

const generateDimension = () =>
  Math.round(Math.random() * ANIMATION_MAX_DIMENSION)

const generateFill = () => {
  const fillColorKey = Math.floor(Math.random() * colors.length)
  return colors[fillColorKey]
}

const generatePosition = (limit: number, dimension: number) => {
  const edge = limit - dimension
  return Math.round(Math.random() * edge)
}

export const generateShapeFactory = (
  canvasHeight: number,
  canvasWidth: number
) => {
  const height = generateDimension()
  const width = generateDimension()

  return {
    fill: generateFill(),
    height,
    width,
    x: generatePosition(canvasWidth, width),
    y: generatePosition(canvasHeight, height),
  }
}
