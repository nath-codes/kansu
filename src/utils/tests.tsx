import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { theme } from '../styles/theme'

export const renderWithTheme = (component: JSX.Element) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
