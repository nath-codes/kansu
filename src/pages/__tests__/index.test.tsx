import 'jest-styled-components'
import React from 'react'
import { renderWithTheme } from '../../utils/tests'
import Index from '../index'

describe('Index', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<Index />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
