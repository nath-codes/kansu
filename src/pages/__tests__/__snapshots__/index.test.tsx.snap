// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
.c1 {
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
  z-index: -1;
}

.c6 {
  align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  display: grid;
  grid-column-gap: 1.125rem;
  grid-template-columns: repeat(3,1.5rem);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  margin: 0;
  color: #ffffff;
}

.c4 {
  display: block;
  height: 3.375rem;
  margin-right: 1.5rem;
}

.c7 {
  position: relative;
}

.c0 {
  display: grid;
  grid-row-gap: 1.5rem;
  grid-template-columns: 90%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: auto;
  max-height: 100vh;
  max-width: 980px;
  overflow: scroll;
  position: relative;
  padding: 0.75rem 0;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 3rem;
  position: relative;
}

.c8:last-child {
  margin-bottom: 3.75rem;
}

.c11 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1.125rem;
  line-height: 1.25;
  margin: auto;
}

.c11 a {
  color: #ffffff;
  position: relative;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c11 a:after {
  border-bottom: solid 2px #ffffff;
  content: "";
  position: absolute;
  bottom: -0.25rem;
  left: 0;
  right: 0;
}

.c11 a:hover {
  border-bottom: solid 2px #49555e;
  color: #49555e;
}

.c11 p {
  margin-bottom: 1.5rem;
}

.c11 p:last-child {
  margin-bottom: 0;
}

.c9 {
  bottom: 0;
  color: #ffffff;
  font-family: "Roboto",sans-serif;
  font-size: 1.5rem;
  left: 0;
  -webkit-letter-spacing: 2.3px;
  -moz-letter-spacing: 2.3px;
  -ms-letter-spacing: 2.3px;
  letter-spacing: 2.3px;
  line-height: 1;
  margin: 0;
  position: absolute;
  text-align: right;
  text-transform: uppercase;
  top: 0.75rem;
  position: absolute;
  writing-mode: sideways-lr;
}

.c9:after {
  color: #ffffff;
  content: "â– ";
  right: 0;
  top: -0.625rem;
  font-size: 0.75rem;
  position: absolute;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-left: 3rem;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  grid-column-gap: 1.5rem;
  grid-row-gap: 1.5rem;
  grid-template-columns: repeat(2,1fr);
  grid-template-rows: 3rem 3rem;
}

.c13 {
  height: 3rem;
  object-fit: contain;
}

@media only screen and (min-width:768px) {
  .c5 {
    grid-column-gap: 2.25rem;
    grid-template-columns: repeat(3,2.25rem);
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    padding: 1.5rem 0;
    grid-row-gap: 3rem;
  }
}

@media only screen and (min-width:1090px) {
  .c0 {
    grid-row-gap: 3.75rem;
    grid-template-columns: 980px;
  }
}

@media only screen and (min-width:768px) {
  .c8 {
    margin-bottom: 4.125rem;
  }

  .c8:last-child {
    margin-bottom: 0;
    padding-bottom: 6.75rem;
  }
}

@media only screen and (min-width:768px) {
  .c11 {
    max-width: 37.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c11 {
    max-width: 42rem;
  }
}

@media only screen and (min-width:768px) {
  .c11 {
    line-height: 1.3333333333;
  }
}

@media only screen and (min-width:768px) {
  .c9 {
    font-size: 1.875rem;
    margin-top: 0.625rem;
  }
}

@media only screen and (min-width:980px) {
  .c9 {
    font-size: 2.25rem;
  }

  .c9:after {
    font-size: 1.5rem;
    top: -1rem;
  }
}

@media only screen and (min-width:768px) {
  .c10 {
    margin-left: 0;
    max-width: 37.5rem;
  }
}

@media only screen and (min-width:980px) {
  .c10 {
    max-width: 40.5rem;
  }
}

@media only screen and (min-width:768px) {
  .c12 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media only screen and (min-width:980px) {
  .c12 {
    grid-column-gap: 3rem;
    grid-row-gap: 3rem;
    grid-template-columns: repeat(3,1fr);
    grid-template-rows: 6rem 6rem;
  }
}

@media only screen and (min-width:980px) {
  .c13 {
    height: 100%;
  }
}

@media only screen and (min-width:768px) {

}

<div
  className="c0"
>
  <div
    className="c1"
    style={
      {
        "filter": "blur(0.1875rem)",
      }
    }
  >
    0
  </div>
  <header
    className="c2"
  >
    <h1
      className="c3"
    >
      <a
        className="c4"
        href="/"
      >
        <img
          alt="Kansu logo"
          src="test-file-stub"
        />
      </a>
    </h1>
    <footer
      className="c5"
    >
      <a
        aria-label="View my projects on Github"
        className="c6"
        href="https://github.com/nathan-kansu"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "fill": "rgba(73, 85, 94, 1)",
          }
        }
      >
        <svg
          viewBox="0 0 33 33"
        >
          <path
            d="M16.608.455C7.614.455.32 7.748.32 16.745c0 7.197 4.667 13.302 11.14 15.456.815.15 1.112-.35 1.112-.783 0-.387-.014-1.41-.022-2.77-4.53.984-5.487-2.184-5.487-2.184-.74-1.882-1.81-2.383-1.81-2.383-1.478-1.01.113-.99.113-.99C7 23.203 7.86 24.767 7.86 24.767c1.454 2.49 3.814 1.77 4.742 1.353.148-1.05.57-1.77 1.034-2.176-3.617-.41-7.42-1.81-7.42-8.05 0-1.78.635-3.234 1.677-4.372-.168-.412-.727-2.07.16-4.31 0 0 1.367-.44 4.48 1.67 1.298-.363 2.692-.543 4.077-.55 1.384.007 2.777.187 4.078.55 3.11-2.11 4.475-1.67 4.475-1.67.89 2.24.33 3.898.163 4.31C26.37 12.66 27 14.115 27 15.892c0 6.26-3.81 7.636-7.437 8.04.584.502 1.105 1.496 1.105 3.015 0 2.178-.02 3.935-.02 4.47 0 .435.294.942 1.12.783 6.468-2.16 11.13-8.26 11.13-15.455 0-8.997-7.293-16.29-16.29-16.29"
          />
        </svg>
      </a>
      <a
        aria-label="Connect with me on LinkedIn"
        className="c6"
        href="https://www.linkedin.com/in/nathan-kansu/"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "fill": "rgba(73, 85, 94, 1)",
          }
        }
      >
        <svg
          viewBox="0 0 521 522"
        >
          <path
            d="M0 66.475V455.55c0 36.725 28.9 65.625 65.625 65.625H454.7c17.975 0 35.15-8.6 46.1-19.525 11.725-11.725 19.525-26.575 19.525-46.1V66.475C520.325 29.75 491.425.85 454.7.85H65.625C46.1.85 31.25 8.675 19.525 20.375 8.575 31.325 0 48.5 0 66.475zm70.325 51.55c0-25 20.325-46.875 45.325-46.875 25.775 0 46.875 21.875 46.875 46.875 0 25.775-21.1 46.1-46.875 46.1-25 0-45.325-20.325-45.325-46.1zm125.775 321.1V200.85c0-5.475 5.475-10.15 9.375-10.15h66.4c9.375 0 9.375 10.95 9.375 17.975 18.75-18.75 42.975-23.45 67.975-23.45 60.95 0 100 28.9 100 92.975v160.95c0 5.475-4.7 10.15-9.375 10.15H371.1c-5.475 0-9.375-5.475-9.375-10.15V293.825c0-24.225-7.025-37.5-34.375-37.5-34.375 0-42.975 22.65-42.975 53.125v129.7c0 5.475-5.475 10.15-10.95 10.15H205.45c-3.9 0-9.375-5.475-9.375-10.15l.025-.025zm-124.225 0V200.85c0-5.475 5.475-10.15 9.375-10.15h67.975c6.25 0 10.15 3.9 10.15 10.15v238.275c0 5.475-4.7 10.15-10.15 10.15H81.25c-4.7 0-9.375-5.475-9.375-10.15z"
            fillRule="nonzero"
          />
        </svg>
      </a>
      <a
        aria-label="Drop me an email"
        className="c6"
        href="mailto:hello@kansu.co"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "fill": "rgba(73, 85, 94, 1)",
          }
        }
      >
        <svg
          viewBox="0 0 486 365"
        >
          <path
            d="M485.21 303.906c0 10.637-2.99 20.498-7.784 29.174l-153.2-171.41 151.54-132.584c5.894 9.355 9.445 20.344 9.445 32.22v242.6zM242.607 192.793L453.47 8.293C444.816 3.556 435.072.65 424.56.65H60.65c-10.523 0-20.27 2.907-28.888 7.644l210.844 184.5zm58.787-11.162l-48.81 42.737c-2.854 2.487-6.41 3.73-9.977 3.73-3.57 0-7.125-1.243-9.98-3.73l-48.82-42.736-155.14 173.6c9.3 5.835 20.198 9.33 31.984 9.33h363.91c11.787 0 22.69-3.495 31.986-9.33l-155.15-173.6zM9.448 29.088C3.554 38.44 0 49.43 0 61.307v242.6c0 10.637 2.978 20.5 7.79 29.174l153.18-171.44L9.448 29.087z"
          />
        </svg>
      </a>
    </footer>
  </header>
  <main
    className="c7"
  >
    <section
      className="c8"
    >
      <h2
        className="c9"
      >
        About
      </h2>
      <div
        className="c10"
      >
        <div
          className="c11"
          style={
            {
              "opacity": 0,
              "transform": " translateY(1.5rem)",
            }
          }
        >
          <p>
            Kansu is myself, Nathan Tredgett. A freelance, senior frontend JavaScript developer based in Bristol, England.
          </p>
          <p>
            With 10 years experience in senior roles, I specialise in React and React-based frameworks, although I have commercial experience in most JS frameworks.
          </p>
          <p>
            My knowledge and experience allows me to consistently deliver effective, well-tested code, quickly. I work with teams of any size to create applications that are fast, consistent and accessible across any and all devices.
          </p>
        </div>
      </div>
    </section>
    <section
      className="c8"
    >
      <h2
        className="c9"
      >
        Clients
      </h2>
      <div
        className="c10"
      >
        <div
          className="c12"
        >
          <img
            alt="Amazon logo"
            className="c13"
            src="test-file-stub"
            style={
              {
                "opacity": 0,
              }
            }
          />
          <img
            alt="Zone Digital logo"
            className="c13"
            src="test-file-stub"
            style={
              {
                "opacity": 0,
              }
            }
          />
          <img
            alt="Mubaloo logo"
            className="c13"
            src="test-file-stub"
            style={
              {
                "opacity": 0,
              }
            }
          />
          <img
            alt="Fresh 8 Logo logo"
            className="c13"
            sizes="(min-width: 320px) 88px, 184px""
            src="test-file-stub"
            srcSet="test-file-stub 88w, test-file-stub 184w, test-file-stub 2x"
            style={
              {
                "opacity": 0,
              }
            }
          />
          <img
            alt="Genomics England logo"
            className="c13"
            sizes="(min-width: 320px) 88px, 184px""
            src="test-file-stub"
            srcSet="test-file-stub 88w, test-file-stub 184w, test-file-stub 2x"
            style={
              {
                "opacity": 0,
              }
            }
          />
          <img
            alt="RHE logo"
            className="c13"
            sizes="(min-width: 320px) 88px, 184px""
            src="test-file-stub"
            srcSet="test-file-stub 88w, test-file-stub 184w, test-file-stub 2x"
            style={
              {
                "opacity": 0,
              }
            }
          />
        </div>
      </div>
    </section>
    <section
      className="c8"
    >
      <h2
        className="c9"
      >
        Contact
      </h2>
      <div
        className="c10"
      >
        <div
          className="c11"
          style={
            {
              "opacity": 0,
              "transform": " translateY(1.5rem)",
            }
          }
        >
          <p>
            I am available for fully-remote contracts, or for hybrid-based contracts in the UK. If you would like to get in touch to discuss any opportunities, please contact me via
             
            <a
              aria-label="Drop me an email"
              href="mailto:hello@kansu.co"
            >
              email
            </a>
            .
          </p>
          <p>
            Alternatively, you can contact me on
             
            <a
              aria-label="Connect with me on LinkedIn"
              href="https://www.linkedin.com/in/nathan-kansu/"
            >
              LinkedIn
            </a>
             
            or view examples of my work via my
             
            <a
              aria-label="View my projects on Github"
              href="https://github.com/nathan-kansu"
            >
              Github
            </a>
            .
          </p>
        </div>
      </div>
    </section>
  </main>
</div>
`;
